styx-service-common/src/main/java/com/spotify/styx/api/Authenticator.java

```diff
@@ -153,6 +157,17 @@ GoogleIdToken authenticate(String token) {
       return null;
     }
 
+    // TODO: Also verify audience for user tokens. Currently this would require changing the auth flow in styx
+    //  clients and make users explicitly "log in" to Styx via the OAuth consent screen.
+    // Verify that this ID token was intended for Styx.
+    if (!allowedAudiences.isEmpty()
+        // TODO: Remove this null check and require tokens to have a target audience
+        && googleIdToken.getPayload().getAudience() != null) {
+      if (!googleIdToken.verifyAudience(allowedAudiences)) {
+        return null;
```

Comment:
Can we log something here before returning? like we do on lines 126, 136, 143,147 and 175.

We don't log any before returning on lines 152, 157, 167, 181 and 184. So it is inconsistent right now but logging here could help us to identify why we are rejecting the authentication